core/remote_state_verifier.go

```diff
@@ -126,16 +143,21 @@ func (vm *remoteVerifyManager) mainLoop() {
 					close(task.terminalCh)
 				}
 			}
+			vm.taskLock.Unlock()
 		case message := <-vm.messageCh:
+			vm.taskLock.RLock()
 			if vt, ok := vm.tasks[message.verifyResult.BlockHash]; ok {
 				vt.messageCh <- message
 			}
+			vm.taskLock.RUnlock()
 
 		// System stopped
 		case <-vm.bc.quit:
+			vm.taskLock.RLock()
```

Comment:
In this loop go routine, we only need write lock before operating vm.tasks, the read lock we can skip?